conda-store-server/conda_store_server/server/app.py

```diff
@@ -340,16 +349,53 @@ async def favicon():
                 name="static-storage",
             )
 
+        @app.on_event("startup")
+        async def startup_event():
+            import signal
+            import time
+
+            from conda_store_server import orm
+
+            # Makes the loop below exit on Ctrl-C
+            signal.signal(signal.SIGINT, sys.exit)
+
+            # Colors
+            green = "\x1b[32m"
+            red = "\x1b[31m"
+            reset = "\x1b[0m"
+
+            # Waits in a loop for worker to become initialized
+            while True:
+                with self.conda_store.session_factory() as db:
+                    q = db.query(orm.Worker).first()
+                    if q is not None and q.initialized:
+                        logger.info(f"{green}" "Worker initialized" f"{reset}")
+                        break
+
+                time.sleep(5)
+                logger.critical(
+                    f"{red}"
+                    "Waiting for worker... "
+                    "Use --standalone if running outside of docker"
+                    f"{reset}"
+                )
```

Comment:
Added a signal handler because uvicorn ignores all signals in the start event. You wouldn't be able to terminate this loop otherwise.

Reply:
Also add a comment for this